shard.go

```diff
@@ -28,22 +31,26 @@ func (s *cacheShard) get(key string, hashedKey uint64) ([]byte, error) {
 
 	if itemIndex == 0 {
 		s.lock.RUnlock()
+		s.miss()
 		return nil, notFound(key)
 	}
 
 	wrappedEntry, err := s.entries.Get(int(itemIndex))
 	if err != nil {
 		s.lock.RUnlock()
+		s.miss()
 		return nil, err
 	}
 	if entryKey := readKeyFromEntry(wrappedEntry); key != entryKey {
 		if s.isVerbose {
 			s.logger.Printf("Collision detected. Both %q and %q have the same hash %x", key, entryKey, hashedKey)
 		}
 		s.lock.RUnlock()
+		s.miss()
```

Comment:
I'm not sure this is a miss as much as it's a legitimate error. Maybe skip this one?

Reply:
Make sense, would like to hear opinions from other contributors.

Reply:
How about adding a dedicated metric for collisions? This might be useful for experiments with different hashing algorithms.